### DISCUSSION TITLE
How many ways to use esp3d in real distanced way?

### DISCUSSION THREAD

**Discussion by Zombie1995** [luc-github/ESP3D#612] [Questions]
How many ways to use esp3d in real distanced way through the internet, to get control over printer really far? Could you please name that methods?

**Comment by luc-github** [comment]
Sorry can you elaborate more your question ?
 using internet there is not limit when you have internet access - but ping can affect performances

**Reply by Zombie1995** [reply]
To control printer, i think, in stack i need to use local ip's. How could I use it through the internet?

**Reply by luc-github** [reply]
you need to ask  your  router / internet access router to do the forwarding - this depend of your internet provider /network and devices
https://codeburst.io/enable-public-access-to-connected-devices-in-your-local-network-c15338da5a3

Have toons of tutorial on google

**Reply by luc-github** [reply]
this is out of scope of ESP3D but be aware doing this you may bring some vulnerability to your network - so do it carefully

**Reply by Zombie1995** [reply]
Thank you!

**Comment by Legsmaniac** [comment]
@Zombie1995 I have set up my ESP3D to enable me to access my ESP3D (and therefore my 3D Printer) remotely.

I did it as @luc-github explains, by configuring my router to allow **Port Forwarding** which means I had to set my own port number in ESP3D which by default uses port 80.  You do NOT want to port forward port 80 as that is VERY risky (most routers won't allow this anyway) so I set my own port number in ESP3D then allowed Port Forward for this number to my ESP3D which I set to a **Static IP Address**.  This was my choice as most new routers allow you to port forward to a **MAC Address** instead which means you don't have to set ESP3D to a Static IP but instead use its MAC Address instead.  As Luc explains, opening a port, ANY port, to the World Wide Web is **not without risk** so for this reason, I also uncommented `#define AUTHENTICATION_FEATURE` in `Configuration.h` and self-compiled to give me the ability to set a Password to access ESP3D, affording me some protection.  Unfortunately, there is no way at the moment to turn on Authentication feature in the ESP3D settings without first uncommenting the line above and recompiling.  _Perhaps @luc-github would consider making this available as an option in the settings in the future?_

I then use a **DDNS Client** to access my own Internet/Router since my IP is dynamic.  If you have a static IP address, you could skip this step.  The router then port forwards the request to my ESP3D.  I have also created my own Android App to use for ESP3D which is simply a webpage to the static IP address of my ESP3D.

Hope this helps.    😉

**Reply by luc-github** [reply]
@Legsmaniac actually I have hardcoded `AUTHENTICATION_FEATURE`  to avoid to introduce a way to disable it, so it is a design choice, currently esp3d  is not really full proof - so better not to add easy back doors, ^_^
I hope to add https support soon but still really busy with webui rewrite now, as https support need several code update especially on esp32 ...

**Reply by Zombie1995** [reply]
Thank you too!

**Comment by Legsmaniac** [comment]
@luc-github I thought that was just to turn off authentication requirement on the serial port?  Something I originally enquired about.

What I meant is, unless #define AUTHENTICATION_FEATURE is uncommented, the Auth option isn't there.  In other words, if someone just grabs the .bin file off your Git and installs that, there's no Auth feature and no option in settings to turn it on.  That would probably be handy for noob users who may wish to use Auth.

**Reply by luc-github** [reply]
I do not provide bin ;)

**Comment by Legsmaniac** [comment]
Ah, no, that's true, you don't.  Tut tut!  Very remiss of you!  🤣
